.github/workflows/quic.yml

```diff
@@ -117,40 +116,82 @@ jobs:
       ref: ${{ github.event.client_payload.sha || github.event.client_payload.ref || inputs.ref || 'main' }}
 
   #
-  # Test Jobs
+  # Manage Azure VMs, TODO: blocks if resources are not available. Waits for it to become available.
+  #
+  manage-azure-vms:
+    name: Manage Azure VMs
+    runs-on: windows-latest
+    outputs:
+      full-matrix: ${{ steps.generate-full-matrix.outputs.matrix }}
+      azure-matrix: ${{ steps.generate-azure-matrix.outputs.matrix }}
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Install Latest Az PowerShell Module
```

Comment:
This is only necessary (I think) when creating the new VMs, because we use some new API not in the default GitHub action VM. Please try removing this everywhere else.

Reply:
The manage-azure-vms script also references the remove-azure-machines script, which does callouts to Azure APIs to remove VMs.